samples/sample_common/src/sample_utils.cpp

```diff
@@ -66,6 +65,71 @@ mfxStatus CopyBitstream2(mfxBitstream *dest, mfxBitstream *src)
     return MFX_ERR_NONE;
 }
 
+mfxU32 GetFrameLength(mfxFrameSurface1* const pSurface, mfxU32 ColorFormat)
+{
+    mfxFrameInfo& info = pSurface->Info;
+    mfxU32 nBytesPerPixel = (info.FourCC == MFX_FOURCC_P010 || info.FourCC == MFX_FOURCC_P210) ? 2 : 1;
+    mfxU16 w = info.CropW ? info.CropW : info.Width;
+    mfxU16 h = info.CropH ? info.CropH : info.Height;
+
+    switch(info.FourCC)
+    {
+    case MFX_FOURCC_YUY2:
+    case MFX_FOURCC_RGB4:
+    case MFX_FOURCC_BGR4:
+    case MFX_FOURCC_AYUV:
+    case MFX_FOURCC_A2RGB10:
+#if (MFX_VERSION >= 1027)
+    case MFX_FOURCC_Y210:
+    case MFX_FOURCC_Y410:
+#endif 
+        switch (ColorFormat)
```

Comment:
from common sense this function should use only one fourcc, which reflects color format of input stream

Reply:
i checked this moment, for correct reading reader need to know both input and encode fourcc (m_ColorFormat is FileInputFourCC, pInfo.FourCC in surface is EncodeFourCC)

Reply:
you need both for proper placing data to surface. You don't need to know fourcc of surface to calculate size of frame in file. This function only calculate frame size